airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py

```diff
@@ -168,10 +169,17 @@ def read(logger, shell_command, is_message=(lambda x: True), transform=(lambda x
             sel.register(p.stderr, selectors.EVENT_READ)
             ok = True
             while ok:
-                for key, val1 in sel.select():
+                for key, _ in sel.select():
                     line = key.fileobj.readline()
                     if not line:
                         ok = False
+                        graceful_wait(p, 60)
+                        if not p.returncode:
+                            raise Exception(f"Underlying command {shell_command} is hanging")
```

Comment:
Wouldn't this read better as saying it's hanging when it sees `subprocess.TimeoutExpired`?

Not using the bool response from  `graceful_wait` and looking at the return code feels strange.

Reply:
you're right, this is here because of an earlier iteration where it made more sense. Removing.